File: backend/register_institution.php
<?php
include 'db.php';

header('Content-Type: application/json');

if ($_SERVER["REQUEST_METHOD"] !== "POST") {
    echo json_encode(["status" => "error", "message" => "Invalid request method!"]);
    exit();
}

// Debug: Log raw received data
$rawData = file_get_contents("php://input");
error_log("Received Data: " . $rawData); // Log to PHP error log

$data = json_decode($rawData, true);

if (!$data) {
    echo json_encode(["status" => "error", "message" => "Invalid JSON format or empty data!", "received" => $rawData]);
    exit();
}

// Check required fields
if (!isset($data["institution_name"], $data["email"], $data["password"], $data["location"], $data["name"])) {
    echo json_encode(["status" => "error", "message" => "All fields are required!", "received" => $data]);
    exit();
}

$institution_name = $conn->real_escape_string($data["institution_name"]);
$admin_name = $conn->real_escape_string($data["name"]);
$email = $conn->real_escape_string($data["email"]);
$password = password_hash($data["password"], PASSWORD_BCRYPT);
$location = $conn->real_escape_string($data["location"]);

$sqlInstitution = "INSERT INTO institutions (institution_name, email, password, location) VALUES ('$institution_name', '$email', '$password', '$location')";

if ($conn->query($sqlInstitution) === TRUE) {
    $institution_id = $conn->insert_id;

    $sqlUser = "INSERT INTO users (name, email, password_hash, institution_id, role) 
                VALUES ('$admin_name', '$email', '$password', '$institution_id', 'admin')";

    if ($conn->query($sqlUser) === TRUE) {
        echo json_encode(["status" => "success", "message" => "Institution and admin registered successfully!"]);
    } else {
        echo json_encode(["status" => "error", "message" => "Institution registered, but failed to create admin: " . $conn->error]);
    }
} else {
    echo json_encode(["status" => "error", "message" => "Failed to register institution: " . $conn->error]);
}

$conn->close();

?>

File: streamlit_app/pages/register_institution.py
import streamlit as st
import requests
import json

API_URL = "http://localhost/event_recommendation/backend/register_institution.php"

st.title("Register Your Institution")

with st.form("register_institution_form"):
    institution_name = st.text_input("Institution Name")
    admin_name = st.text_input("Admin Name")
    email = st.text_input("Email")
    password = st.text_input("Password", type="password")
    location = st.text_input("Location")

    submit_button = st.form_submit_button("Register Institution")

if submit_button:
    if not institution_name or not admin_name or not email or not password or not location:
        st.error("⚠️ All fields are required!")
    else:
        data = {
            "institution_name": institution_name.strip(),
            "name": admin_name.strip(),
            "email": email.strip(),
            "password": password.strip(),
            "location": location.strip()
        }

        headers = {'Content-Type': 'application/json'}

        # 🚀 Make sure to send data as JSON
        response = requests.post(API_URL, data=json.dumps(data), headers=headers)  

        try:
            result = response.json()
          

            if response.status_code == 200 and result["status"] == "success":
                st.success(result["message"])
            else:
                st.error(f"❌ {result.get('message', 'Registration failed!')}")
        except requests.exceptions.JSONDecodeError:
            st.error("This institution already exists")
